kernel: bump 6.6 to 6.6.89

namiltd opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.89

All patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

---

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.88

Manually rebased:
 - bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch[1]
 - bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch[2]
 - generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch[3]
 - generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch[4]

All other patches automatically rebased.

1. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=f249c05416ea0bef24c9dbed0e653d2fad87b127
2. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=1fea7726276e5d6526ecd4e7ccb4c91a6135deb5
3. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=95f17738b86fd198924d874a5639bcdc49c7e5b8
4. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=a2874f0dff63829d1f540003e2d83adb610ee64a

graysky2 commented:
```
Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64
```

DragonBluep commented on target/linux/generic/pending-6.6/620-net_sched-codel-do-not-defer-queue-length-update.patch:
```diff
@@ -68,14 +65,12 @@ Link: https://bugzilla.kernel.org/show_bug.cgi?id=109581
--	 */
--	if (q->cstats.drop_count && sch->q.qlen) {
+-
+-	if (q->cstats.drop_count) {
```
LGTM for this patch. By the way, it will be removed in https://github.com/openwrt/openwrt/pull/18618.

DragonBluep reviewed (commented):
@qosmio had already created the 6.6.88 kernel bump patch two days ago https://github.com/openwrt/openwrt/pull/18604. I would suggest adding a `Co-authored-by` tag.
```
From c7ece9a3b5b9049b3dbbf0354989edc19a6f83e0 Mon Sep 17 00:00:00 2001
From: Sean Khan <datapronix@protonmail.com>
Date: Fri, 25 Apr 2025 22:55:42 -0400
Subject: [PATCH] kernel: bump 6.6 to 6.6.88
```

namiltd commented on target/linux/bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch:
```diff
@@ -14,13 +14,13 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
+-	ov7251->enable_gpio = devm_gpiod_get(dev, "enable", GPIOD_OUT_LOW);
 +	ov7251->enable_gpio = devm_gpiod_get_optional(dev, "enable",
-+						      GPIOD_OUT_HIGH);
++						      GPIOD_OUT_LOW;
```
Fixed

namiltd commented:
Update 950-0521-PCI-brcmstb-Add-BCM2712-support.patch in progress

namiltd commented:
@DragonBluep I updated 950-0521-PCI-brcmstb-Add-BCM2712-support.patch
Could you take a look at this?

danpawlik commented:
Works well on BPI-R4 and GL.iNet B-3000.

DragonBluep commented on target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch:
```diff
@@ -1019,41 +1019,45 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
++				dev_err(pcie->dev, "Unable to find MSI PCI address\n");
++				ret = -EINVAL;
++				goto fail;
++		}	
```
Missing a tab here.

DragonBluep commented on target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch:
```diff
@@ -1019,41 +1019,45 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
+ 
+-		if (msi_np == pcie->np)
++		if (msi_np == pcie->np) {
+ 			ret = brcm_pcie_enable_msi(pcie);
```
```suggestion
 			ret = brcm_pcie_enable_msi(pcie);
+			of_node_put(msi_np);
```

DragonBluep commented on target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch:
```diff
@@ -1019,41 +1019,45 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
++
++			if (of_property_read_u64(msi_np, "brcm,msi-pci-addr", &msi_pci_addr)) {
++				dev_err(pcie->dev, "Unable to find MSI PCI address\n");
++				ret = -EINVAL;
```
```suggestion
+				ret = -EINVAL;
+				of_node_put(msi_np);
```

DragonBluep commented on target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch:
```diff
@@ -1019,41 +1019,45 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
++
++			if (of_property_read_u64(msi_np, "reg", &msi_phys_addr)) {
++				dev_err(pcie->dev, "Unable to find MSI physical address\n");
++				ret = -EINVAL;
```
```suggestion
+				ret = -EINVAL;
+				of_node_put(msi_np);
```

DragonBluep commented on target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch:
```diff
@@ -1019,41 +1019,45 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
++				ret = -EINVAL;
++				goto fail;
++			}
+ 
```
```suggestion
 
 +			of_node_put(msi_np);
```

DragonBluep commented on target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch:
```diff
@@ -1019,41 +1019,45 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
-+		writel(upper_32_bits(msi_phys_addr),
-+		       pcie->base + PCIE_MISC_UBUS_BAR1_CONFIG_REMAP_HI);
- 	}
+ 		of_node_put(msi_np);
```
remove this `of_node_put(msi_np);`

DragonBluep reviewed (commented):
> @DragonBluep I updated bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
Could you take a look at this?

We need to call `of_node_put(msi_np);` before return in patch 950-0521.

Edit: remove some unnecessary suggestions.

pesa1234 reviewed (approved):
```
Build-tested: GL.iNet GL-MT6000
Run-tested: GL.iNet GL-MT6000
```

schuettecarsten commented:
Works on bcm27xx/bcm2711 (rpi4)

graysky2 commented:
```
Build system: x86/64
Build-tested: GL.iNet GL-MT6000
Run-tested: GL.iNet GL-MT6000
```

namiltd committed:
kernel: bump 6.6 to 6.6.88

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.88

Manually rebased:
 - bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch[1]
 - bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch[2]
 - generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch[3]
 - generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch[4]

All other patches automatically rebased.

1. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=f249c05416ea0bef24c9dbed0e653d2fad87b127
2. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=1fea7726276e5d6526ecd4e7ccb4c91a6135deb5
3. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=95f17738b86fd198924d874a5639bcdc49c7e5b8
4. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=a2874f0dff63829d1f540003e2d83adb610ee64a

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/18607
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

stklcode commented:
Build and run tested for mvebu/cortexa9 (cznic_turris-omnia)

rockdrilla commented:
@namiltd may I kindly ask you to update to 6.6.89? Thanks in advance.

graysky2 committed:
kernel: bump 6.6 to 6.6.89

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.89

All patches automatically rebased.

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/18607
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

graysky2 commented:
@namiltd - I edited the commit msg to include the tests/runs:


```    
    Build system: x86/64
    Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
    Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
```
